test/visibility/private/uses/weirdEnumHiding.chpl

```diff
@@ -24,14 +24,14 @@ module weirdEnumHiding {
     public use super.C;
     public use super.B;
 
-    proc checkFoo() {
+    proc checkFooD() {
       writeln(foo.b);
     }
   }
 
   proc main() {
-    B.checkFoo();
-    C.checkFoo();
-    D.checkFoo();
+    B.checkFooB();
+    C.checkFooC();
+    D.checkFooD();
```

Comment:
I may just be fried here at file 85, but I'm not following this one.  I think the renaming is necessary because the `public use`s of super.C and super.B conflict with one another?  Yet nothing uses module `D` so are those public uses even required — could we just remove them and avoid the renaming?

I'm also not clear enough on what bug this was trying to keep locked out of the compiler to know whether either the current approach or mine would keep the bug at bay (or whether the bug is moot with the changes being proposed in this PR).  Probably a question for @lydia-duncan ?

Reply:
Right, I checked with @lydia-duncan about this off-github and she said that the access to the enum was the important part so renaming the checkFoo functions to have different names was a good way to resolve the problem.

>  I think the renaming is necessary because the public uses of super.C and super.B conflict with one another? 

Yes, and also with `proc checkFoo()` in D.

> Yet nothing uses module D so are those public uses even required — could we just remove them and avoid the renaming?

I think that is true but I wanted to leave the public uses there to keep the test closer to whatever bug case it is trying to keep at bay.

Reply:
Nothing uses `D` *in this file*, there is a separate test file that uses it.  Here's the information I provided about the test when reviewing this PR earlier:

> The test was originally added to support https://github.com/chapel-lang/chapel/blob/main/test/visibility/private/uses/use_weirdEnumHiding.chpl in https://github.com/chapel-lang/chapel/pull/13355 - I was seeing a bug where private use A -> public use B was causing us to skip later uses of B in the same scope. With that in mind, I do think they need to stay public.
>
> Additionally, iirc we already have tests to ensure that more limited use statements (`use A only x;`, `use A except bar;`, etc) don't prevent later unlimited uses of the same module (`use A;`), so I think if we don't think this should be possible, it should either be removed as a test or changed to track a failure mode.

Basically, we weren't recognizing early enough that a use was private and so accidentally were exposing uses and imports in its scope and letting that impact whether we looked into those modules later.

----

Here's some additional thoughts I provided in the earlier thread, since Brad was similarly confused about what caused the need for the qualified access:

> It took me a while to figure out why this change was necessary, since all checkFoos were used in a qualified fashion. It seems like this effectively neuters qualified access to symbols in D that might conflict with other things, and that worries me.

And:

> >  By having a public use, module D is saying it wants to re-export C.checkFoo and B.checkFoo but it can't do that (because of the duplicate symbol between C and B and because D also defines the same thing).

> It's saying that it both/either wants to re-export C.checkFoo as C.checkFoo or checkFoo. With a public use, you could write D.C.checkFoo to get at that symbol. If D didn't define its own checkFoo, you'd also be able to write D.checkFoo or checkFoo (if it didn't also re-export from B). The distinction matters less right now because all three functions take no arguments - where being able to explicitly choose the one defined by D itself matters is when it's less clear which version should be called, such as when coercion on the argument type is possible. If a user wanted to explicitly ensure they always used each particular version, regardless of if we changed language level prioritization of coercion, they could use qualified access to do it. Or they would be able to without this change.